librosa/beat.py

```diff
@@ -371,12 +371,14 @@ def tempo(
     if max_tempo is not None:
         max_idx = np.argmax(bpms < max_tempo)
         logprior[:max_idx] = -np.inf
+    # explicit axis expansion
+    logprior = np.expand_dims(logprior, axis=(-1, *range(tg.ndim-2)))
```

Comment:
Fancy :monocle_face:

But this relies on deprecated functionality in expand_dims when `tg.ndim-2 > logprior.ndim`.  See the deprecation note [here](https://numpy.org/doc/stable/reference/generated/numpy.expand_dims.html).